src/sqlfluff/core/rules/base.py

```diff
@@ -494,19 +509,24 @@ def eval(self, **kwargs):
         :exc:`ValueError`.
 
         """
-        elems = cls.__name__.split("_")
+        rule_name_match = self.valid_rule_name_regex.match(cls.__name__)
         # Validate the name
-        if len(elems) != 2 or elems[0] != "Rule" or len(elems[1]) != 4:
+        if not rule_name_match:
             raise ValueError(
                 (
-                    "Tried to register rule on set {0!r} with unexpected " "format: {1}"
-                ).format(self.name, cls.__name__)
+                    "Tried to register rule on set {0!r} with unexpected "
+                    "format: {1}, format should be: {2}."

```

Comment:
Nitpick: In the error message, should we consider providing a human-readable version of the naming pattern?